runtime/caml/signals.h

```diff
@@ -34,14 +34,85 @@ CAMLextern void caml_enter_blocking_section (void); // raises
+   they do not raise or fail, but also that they are brief (typically,
+   systhread's yield()), because themselves cannot be interrupted.
+
+   In contrast, asynchronous callbacks of the second kind are allowed
+   to raise an exception (typically, when handling SIGINT or SIGALRM),
+   are not limited in what they can express (typically, finalisers and
+   memprof callbacks), and can themselves be interrupted by
+   asynchronous callbacks.
+
+   Masking temporarily delays the execution of asynchronous callbacks.
+   There are two kinds of masking:
+
+   - The _nonpreemptible mask_ delays execution of all asynchronous
+     callbacks.
+
+   - The _uninterruptible mask_ delays the execution of unrestricted
+     asynchronous callbacks, whereas
+
+   This gives three possible values for
+   [Caml_state->mask_async_callbacks]: CAML_MASK_NONE,
+   CAML_MASK_UNINTERRUPTIBLE and CAML_MASK_NONPREEMPTIBLE.
+
+   In particular:
+
+   - a program in a nonpreemptible mask will not run any OCaml code
+     asynchronously or concurrently;
+
+   - a program in an uninterruptible mask will not be interrupted by
+     asynchronous exceptions.
+
+*/
+
+CAMLextern caml_mask_kind caml_mask (caml_mask_kind new_mask);
+CAMLextern           void caml_unmask (caml_mask_kind old_mask);
+/* The caml_mask function sets an uninterruptible or nonpreemptible
+   mask and returns the previous mask. The caml_unmask function sets
+   the previous mask back. Assumes that the runtime lock is held.
+
+   The caml_mask function never undoes a previous mask, that is,
+   calling caml_mask(CAML_MASK_UNINTERRUPTIBLE) has no effect inside an
+   nonpreemptible mask. Similarly, caml_mask(CAML_MASK_NONE) has no
+   effect at all, it thus can be used to query the current mask.
+
+   Calls to caml_mask and caml_unmask must be correctly bracketed and
+   caml_unmask must be supplied value returned from the corresponding
+   call to caml_mask (the previous mask). It is incorrect to call
+   caml_unmask with a different mask, so as to undo a mask one does
+   not own.
+
+   Example:
+
+     caml_mask_kind old_mask = caml_mask(CAML_MASK_UNINTERRUPTIBLE);
+     ...(no raising of exception)...
+     caml_unmask(old_mask);
+*/
+
 CAMLextern void caml_process_pending_actions (void);
-/* Checks for pending actions and executes them. This includes pending
-   minor and major collections, signal handlers, finalisers, and
-   Memprof callbacks. Assumes that the runtime lock is held. Can raise
-   exceptions asynchronously into OCaml code. */
+/* Checks for pending actions and executes them, depending on the
+   current mask. Assumes that the runtime lock is held.
+
+   Can raise exceptions asynchronously into OCaml code if the current
+   mask is CAML_MASK_NONE.
+*/
 
 CAMLextern int caml_check_pending_actions (void);
-/* Returns 1 if there are pending actions, 0 otherwise. */
+/* Returns 1 if there are actions to do at current mask level, 0
```

Comment:
very nitpicky: breaking the line right after the comma would be nicer

Reply:
I leave it up to my emacs macro...